test/src/main/java/org/apache/accumulo/test/ScanServerIT.java

```diff
@@ -246,6 +255,51 @@ public void testScanWithTabletHostingMix() throws Exception {
     }
   }
 
+  @Test
+  public void testScanTabletsWithOperationIds() throws Exception {
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      String tableName = getUniqueNames(1)[0];
+
+      setupTableWithHostingMix(client, tableName);
+
+      // Unload all tablets
+      TableId tid = TableId.of(client.tableOperations().tableIdMap().get(tableName));
+      client.tableOperations().setTabletHostingGoal(tableName, new Range((Text) null, (Text) null),
+          TabletHostingGoal.ONDEMAND);
+
+      // Wait for the tablets to be unloaded
+      Wait.waitFor(() -> ScanServerIT.getNumHostedTablets(client, tid.canonical()) == 0, 30_000,
+          1_000);
+
+      // Set operationIds on all the table's tablets so that they won't be loaded.
+      TabletOperationId opid = TabletOperationId.from(TabletOperationType.SPLITTING, 1234L);
+      Ample ample = getCluster().getServerContext().getAmple();
+      ServerAmpleImpl sai = (ServerAmpleImpl) ample;
+      try (TabletsMutator tm = sai.mutateTablets()) {
+        ample.readTablets().forTable(tid).build().forEach(meta -> {
+          tm.mutateTablet(meta.getExtent()).putOperation(opid).mutate();
+        });
+      }
+
+      try (Scanner scanner = client.createScanner(tableName, Authorizations.EMPTY)) {
+        scanner.setRange(new Range());
+        // Confirm that the ScanServer will not complete the scan
+        scanner.setConsistencyLevel(ConsistencyLevel.EVENTUAL);
+        assertThrows(AssertionFailedError.class,
+            () -> assertTimeoutPreemptively(Duration.ofSeconds(30), () -> {
+              Iterables.size(scanner);
+            }));
+
+        // Confirm that the TabletServer will not complete the scan
+        scanner.setConsistencyLevel(ConsistencyLevel.IMMEDIATE);
+        assertThrows(AssertionFailedError.class,
+            () -> assertTimeoutPreemptively(Duration.ofSeconds(30), () -> {
+              Iterables.size(scanner);
+            }));
+      } // when the scanner is closed, all open sessions should be closed
```

Comment:
Seems like this code will wait 30 seconds and then wait 30 seconds.  Maybe could refactor the code to do something like the following.

 1. create executor w/ two threads
 2. future1 = executor.submit( eventual scan )
 3. future2 = executor.submit( immediate scan )
 4. wait 30 secs
 5. assert futures are not done and cancel them

This allows the scans to run concurrently and only have a single wait.

Reply:
Could take things a bit further and combine the batch scan and regular scan with something like the following

 1. create executor w/ four threads
 2. future1 = executor.submit( eventual scan )
 3. future2 = executor.submit( immediate scan )
 4. future3 = executor.submit( eventual batch scan )
 5. future4 = executor.submit( immediate batch scan )
 6. wait 30 secs
 7. assert futures are not done and cancel them

Reply:
Implemented in 019cd50